src/app/components/configure-app/configure-app.component.ts

```diff
@@ -300,12 +316,30 @@ export class ConfigureAppComponent implements OnInit {
         this.notifications.sendWarning(`CPU Worker support not available, set PoW to Best`);
         newPoW = 'best';
       }
+      // reset multiplier when not using it to avoid user mistake
+      if (newPoW !== 'clientWebGL' && newPoW !== 'clientCPU') {
+        this.selectedMultiplierOption = this.multiplierOptions[0].value;
+      }
+    }
+
+    // reset work cache so that the new PoW will be used but only if larger than before
+    if (
+      newMultiplier > this.appSettings.settings.multiplierSource &&
+      newMultiplier > 1 &&
+      ((newPoW === 'clientWebGL' && this.pow.hasWebGLSupport()) ||
+      (newPoW === 'clientCPU' && this.pow.hasWorkerSupport()))
+    ) {
+      // if user accept to reset cache
+      if (await this.clearWorkCache()) {
+        reloadPending = true; // force reload balance => re-work pow
+      }
     }
 
     const newSettings = {
       walletStore: newStorage,
       lockInactivityMinutes: Number(this.selectedInactivityMinutes),
       powSource: newPoW,
+      multiplierSource: Number(this.selectedMultiplierOption),
```

Comment:
Why do we need to cast `this.selectedMultiplierOption` to Number? Isn't it already set to number in line 123?

Reply:
Yeah that was strange. Without the Number, it got converted to string in the local store. Casting to Number solved the issue